pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -158,10 +160,44 @@ func (w *Watcher) invokeRegistrationCallbackAtHandler(ctx context.Context, clien
 	}); err != nil {
 		return fmt.Errorf("failed to send registration status at socket %s, err: %v", socketPath, err)
 	}
+
 	chanForAckOfNotification <- true
 	return nil
 }
 
+// Handle filesystem notify event.
+func (w *Watcher) handlefsnotifyEvent(event fsnotify.Event) error {
+	if event.Op&fsnotify.Create != fsnotify.Create {
+		return nil
+	}
+
+	fi, err := os.Stat(event.Name)
+	if err != nil {
+		return fmt.Errorf("stat file %s failed: %v", event.Name, err)
+	}
+
+	if !fi.IsDir() {
```

Comment:
If the path is not a dir, is there further check that can be done? Is it assume if it's a file it's socket file ?

Reply:
The only check that we've been doing is checking that it doesn't start with a '.'
More generally you won't know if it's a gRPC server until you try to connect to it

Reply:
@RenaudWasTaken if it's a regular file, then you know you cannot connect it, right?

Reply:
i added file mode check in addition to Dir check, thanks
https://golang.org/pkg/os/#FileMode